### PR TITLE
Kmp-ify testutils-ktx

### PR DESCRIPTION
**Pull request by veyndan**
Upstreaming [Multiplatform Paging's kmp-ified testutils-ktx](https://github.com/cashapp/multiplatform-paging/tree/androidx-main-3.2.0-alpha04/testutils/testutils-ktx). See https://github.com/androidx/androidx/pull/505#issuecomment-1485536890 for more context.

Test: ./gradlew test connectedCheck
Bug: 270612487

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -255,6 +255,7 @@ skikoLinuxArm64 = { module = "org.jetbrains.skiko:skiko-awt-runtime-linux-arm64"
 sqldelightAndroid = { module = "com.squareup.sqldelight:android-driver", version.ref = "sqldelight" }
 sqldelightCoroutinesExt = { module = "com.squareup.sqldelight:coroutines-extensions", version.ref = "sqldelight" }
 sqliteJdbc = { module = "org.xerial:sqlite-jdbc", version = "3.36.0" }
+statelyIsoCollections = { module = "co.touchlab:stately-iso-collections", version = "1.2.0" }
```
Comment (ianhanniballake) [inline gradle/libs.versions.toml]: From the [comments here](https://github.com/touchlab/Stately/tree/main/deprecated#stately-iso-collections), it seems like this library is completely deprecated and seems to say that the `ArrayDeque` that is now part of the Kotlin Standard Library should be enough. Do all the tests still pass if we swap to the stdlib version?

Comment (veyndan) [reply gradle/libs.versions.toml]: >  […] it seems like this library is completely deprecated […]

The library isn't deprecated, but the module is. Unfortunately, the [proposed replacement (`stately-concurrent-collections`)](https://github.com/touchlab/Stately#stately-concurrent-collections) for this module [isn't stable yet](https://mvnrepository.com/artifact/co.touchlab/stately-concurrent-collections).

> […] and seems to say that the `ArrayDeque` that is now part of the Kotlin Standard Library should be enough.

A KMP version of `java.util.ArrayDeque` [was added in Kotlin 1.3.70](https://blog.jetbrains.com/kotlin/2020/03/kotlin-1-3-70-released/#Double-endedqueueimplementation:ArrayDeque). Like `java.util.ArrayDeque`, `kotlin.collections.ArrayDeque` isn't thread-safe.

> Do all the tests still pass if we swap to the stdlib version?

From the previous point, `ArrayQueue` isn't thread-safe. Swapping out `IsoArrayDeque` with `ArrayDeque` _could_ make all the tests pass, but those tests would presumably be a bit flaky.

**Comment by claraf3** [comment]
Hi @veyndan thank you for your patience. After we merged your other PRs removing most usages of TestDispatcher, the remaining tests still using it appears to be single threaded access to the queue. I think its safe to at this point to switch to `ArrayDequeue`. I think this might be a better route for now so we can prevent the negatives of introducing a deprecated 3rd party module.

**Comment by veyndan** [comment]
@claraf3 Sounds good to me! I've just updated the PR to use `ArrayDeque`, but I'm getting a CI error. It seems unrelated to my changes though.

**Comment by claraf3** [comment]
Awesome, thank you Veyndan! Yeah it looks to be our issue. Don't worry about it :)

**Review by claraf3 — APPROVED** [review]

**Review by ianhanniballake — APPROVED** [review]

File: testutils/testutils-ktx/src/jvmMain/kotlin/androidx/testutils/VerifyWithPolling.kt
Comment (ianhanniballake) [inline testutils/testutils-ktx/src/jvmMain/kotlin/androidx/testutils/VerifyWithPolling.kt:1]: We're getting a lint failure on this file:
```
$SUPPORT/testutils/testutils-ktx/src/jvmMain/kotlin/androidx/testutils/VerifyWithPolling.kt:29: Error: Uses Thread.sleep() [BanThreadSleep from androidx.build]
Thread.sleep(periodMs)
~~~~~

Explanation for issues of type "BanThreadSleep":
Use of Thread.sleep() is not allowed, please use a callback or another way
to make more reliable code. See more details at
https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:docs/testability.md#calling-threadsleep-as-a-synchronization-barrier
```

Do you mind seeing if you can suppress the `BanThreadSleep` check here? Ideally you should be able to reproduce this with by running `:internal-testutils-ktx:lint`

Comment (veyndan) [reply testutils/testutils-ktx/src/jvmMain/kotlin/androidx/testutils/VerifyWithPolling.kt:1]: I tried recreating this issue locally, but it always passes. I'm running the following command within the `paging` directory:

```
./gradlew :internal-testutils-ktx:lint --rerun-tasks --no-configuration-cache   
```

Comment (claraf3) [reply testutils/testutils-ktx/src/jvmMain/kotlin/androidx/testutils/VerifyWithPolling.kt:1]: I was able to repro the error on my end. I added the suppression and its running through checks. Should be good now, thanks for trying!

**Pull request closed without merging**